GearRent/ViewControllers/CreateListingViewController.m

```diff
@@ -81,13 +85,26 @@ - (void)viewDidLoad {
     fingerTap.numberOfTapsRequired = 1;
     fingerTap.numberOfTouchesRequired = 1;
     [self.mapView addGestureRecognizer:fingerTap];
+    self.categoriesPicker.delegate = self;
+    self.categoriesPicker.dataSource = self;
+    __weak typeof(self) weakSelf = self;
+    void(^completion)(NSArray<Category *> *categories, NSError *error) = ^void(NSArray<Category*> *categories, NSError *error){
+        typeof(self) strongSelf = weakSelf;
+        if(error == nil){
+            strongSelf.categories = categories;
+            [strongSelf.categoriesPicker reloadAllComponents];
+        } else{
+            NSLog(@"%@", error);
+        }
+    };
+    fetchAllCategories(completion);
     if(self.listing != nil){
         self.navigationItem.title = @"Edit Listing";
         self.deleteListingButton.hidden = NO;
         self.addImagesButton.hidden = YES;
         self.titleTextField.text = self.listing.title;
         self.priceTextField.text = [[NSNumber numberWithFloat: self.listing.price] stringValue];
```

Comment:
You can use the `modern` objective-c format here 

```OBJC
        self.priceTextField.text = [@(self.listing.price) stringValue];
```

However, the best practice is to use `NSNumberFormatter` with a currency style to make sure your numbers are currently formatted across different locales. 

```OBJC
    NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
    formatter.numberStyle = NSNumberFormatterCurrencyStyle;
    NSString *currency = [formatter stringFromNumber:number];
```